### PR TITLE
Implement serialization for OffsetDate and OffsetTime

### PR DESCRIPTION
**Pull request by jskeet**
I think this completes the code changes for #864. We then need to do
documentation, and when 2.3 has been released we can look at
Json.NET support.

### CODE REVIEW

File: src/NodaTime.Test/OffsetDateTest.cs [hunk 1]
```diff
@@ -148,5 +151,39 @@ public void Deconstruction()
...
+        public void BinarySerialization()
+        {
+            var date = new LocalDate(2013, 4, 12);
+            var offset = Offset.FromHoursAndMinutes(5, 30);
+            var value = new OffsetDate(date, offset);
+            TestHelper.AssertBinaryRoundtrip(value);
+        }
+
+        [Test]
+        public void XmlSerialization_Iso()
```
Comment (malcolmr) [inline src/NodaTime.Test/OffsetDateTest.cs]: Could you add tests covering BCE?

Specifically, for OffsetDate, I assume that the serialized form of January 1st, 2 BCE would be something like `-0001-01-01Z`, following (I assume) ISO 8601:2000, and not e.g. the craziness discussed in https://www.w3.org/TR/xmlschema-2/#dateTime.

Comment (malcolmr) [inline src/NodaTime.Test/OffsetDateTest.cs]: Also a test that clarifies whether 123 AD is `123-01-01Z` or (what I believe might be required for `xs:date`) `0123-01-01Z` would be nice.

Comment (jskeet) [reply src/NodaTime.Test/OffsetDateTest.cs]: Done, thanks. (And yes, we're following ISO-8601.)

Comment (jskeet) [reply src/NodaTime.Test/OffsetDateTest.cs]: It's 0123-01-01 - all our default formatters use `uuuu` as the year specifier.

**Pull request merged**